adds missing data to MBODA questionnaire results

raleigh-g-thompson opened the pull request:
adds missing data to the expected results for the MBODA Initial Expressions questionnaire test cases

brynrhodes commented on input/tests/library/MBODAInitialExpressions/USCorePatient-1/Parameters-expected-results.json:
```diff
@@ -22,85 +21,315 @@
-                "resourceType": "Quantity",
-                "value": 176.0,
-                "unit": "cm"
+                "resourceType": "Reference",
```
For a resource-valued expression that is null, can use valueBoolean to communicate the extension (but note the need for a cql-type extension as well to indicate the actual result type

raleigh-g-thompson replied:
Can we use the cqlType extension? Isn't that a FHIR v5 extension, the questionnaire examples are FHIR v4.01. Are we uplifting the questionnaires to v5?

brynrhodes commented on input/tests/library/MBODAInitialExpressions/USCorePatient-1/Parameters-expected-results.json:
```diff
@@ -22,85 +21,315 @@
-                "value": 80.0,
-                "unit": "kg"
+                "resourceType": "Reference",
+                "id": "MedicationRequest/uscore-patient-1-med-request-example"
```
Should be inline resource

brynrhodes commented on input/tests/library/MBODAInitialExpressions/USCorePatient-1/Parameters-expected-results.json:
```diff
@@ -22,85 +21,315 @@
-                "value": 1.9641572,
-                "unit": "m2"
+                "resourceType": "Reference",
+                "id": "MedicationRequest/uscore-patient-1-med-request-example"
```
Should be inline resource

brynrhodes reviewed (changes requested):
Generally, if the result of the CQL expression is a resource, then that would be represented with an in-line resource.

If it's null, valueBoolean, with an extension for data-absent-reason, plus a cqlType extension to indicate the actual result type

raleigh-g-thompson commented:
Addressed issues related to reference values. Converted them to in-line resources, or marked as 'null' using the cqf-cqlType extension and the appropiate cqf-isEmptyList of data-absent extensions.
